pkg/resources/grant_privileges_to_share.go

```diff
@@ -0,0 +1,474 @@
+	//	Description: "The fully qualified name of the function on which privileges will be granted.",
+	//	ValidateDiagFunc: IsValidIdentifier[sdk.FunctionIdentifier](),
+	//	ExactlyOneOf: grantPrivilegesToShareGrantExactlyOneOfValidation,
+	// },
+	"table_name": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ForceNew:         true,
+		Description:      "The fully qualified name of the table on which privileges will be granted.",
+		ValidateDiagFunc: IsValidIdentifier[sdk.SchemaObjectIdentifier](),
+		ExactlyOneOf:     grantPrivilegesToShareGrantExactlyOneOfValidation,
+	},
+	"all_tables_in_schema": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ForceNew:         true,
+		Description:      "The fully qualified identifier for the schema for which the specified privilege will be granted for all tables.",
+		ValidateDiagFunc: IsValidIdentifier[sdk.DatabaseObjectIdentifier](),
+		ExactlyOneOf:     grantPrivilegesToShareGrantExactlyOneOfValidation,
+	},
+	"tag_name": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ForceNew:         true,
+		Description:      "The fully qualified name of the tag on which privileges will be granted.",
+		ValidateDiagFunc: IsValidIdentifier[sdk.SchemaObjectIdentifier](),
+		ExactlyOneOf:     grantPrivilegesToShareGrantExactlyOneOfValidation,
+	},
+	"view_name": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ForceNew:         true,
+		Description:      "The fully qualified name of the view on which privileges will be granted.",
+		ValidateDiagFunc: IsValidIdentifier[sdk.SchemaObjectIdentifier](),
+		ExactlyOneOf:     grantPrivilegesToShareGrantExactlyOneOfValidation,
+	},
+}
+
+func GrantPrivilegesToShare() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateGrantPrivilegesToShare,
+		UpdateContext: UpdateGrantPrivilegesToShare,
+		DeleteContext: DeleteGrantPrivilegesToShare,
+		ReadContext:   ReadGrantPrivilegesToShare,
+
+		Schema: grantPrivilegesToShareSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: ImportGrantPrivilegesToShare(),
+		},
+	}
+}
+
+func ImportGrantPrivilegesToShare() func(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	return func(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+		id, err := ParseGrantPrivilegesToShareId(d.Id())
+		if err != nil {
+			return nil, err
+		}
+		if err := d.Set("share_name", id.ShareName.Name()); err != nil {
+			return nil, err
+		}
+		if err := d.Set("privileges", id.Privileges); err != nil {
+			return nil, err
+		}
+
+		switch id.Kind {
+		case OnDatabaseShareGrantKind:
+			if err := d.Set("database_name", id.Identifier.Name()); err != nil {
+				return nil, err
+			}
+		case OnSchemaShareGrantKind:
+			if err := d.Set("schema_name", id.Identifier.FullyQualifiedName()); err != nil {
+				return nil, err
+			}
+		// case OnFunctionShareGrantKind:
```

Comment:
Add TODO with jira number here please (it will be easier to reference it later)

Reply:
It also affects all the other places in this resource where you commented out the functions-related implementation. We should either remove these or leave a comment with the ticket number on each of them.

Reply:
I added TODO(ticket number) in all of the places someone would need to visit in order to add function_name param